onnxruntime/contrib_ops/cpu/bert/attention_cpu_base.h

```diff
@@ -85,19 +90,29 @@ class AttentionCPUBase : public AttentionBase {
     T* present_key_data = present_key != nullptr ? present_key->MutableData<T>() : nullptr;
     const T* past_value_data = past_value != nullptr ? past_value->Data<T>() : nullptr;
     T* present_value_data = present_value != nullptr ? present_value->MutableData<T>() : nullptr;
+    T* scaled_qk_data = scaled_qk != nullptr ? scaled_qk->MutableData<T>() : nullptr;
 
     const T* attn_bias_data = (attn_bias != nullptr) ? attn_bias->Data<T>() : nullptr;
     auto attn_bias_dims = (attn_bias != nullptr) ? attn_bias->Shape().GetDims() : gsl::span<const int64_t>{};
 
+    // Used for DecoderMaskedMultiHeadAttention
+    int max_sequence_length = 0;
+    if (use_dmmha) {
+      ORT_ENFORCE(past_key != nullptr && past_value != nullptr);
+      max_sequence_length = static_cast<int>(past_key->Shape().GetDims()[2]);
```

Comment:
Here we use past_key buffer length as max_sequence_length.

Do we assume that we have shared buffer for past and present for DecoderMaskedMultiHeadAttention?